pyomo/dae/tests/test_set_utils.py

```diff
@@ -0,0 +1,162 @@
+            def b(bl):
+                bl.v = Var()
+                bl.v1 = Var(m.set)
+                bl.v2 = Var(m.time)
+
+        disc = TransformationFactory('dae.collocation')
+        disc.apply_to(m, wrt=m.time, nfe=5, ncp=2, scheme='LAGRANGE-RADAU')
+        disc.apply_to(m, wrt=m.space, nfe=5, ncp=2, scheme='LAGRANGE-RADAU')
+
+        self.assertFalse(is_explicitly_indexed_by(m.v, m.time))
+        self.assertTrue(is_explicitly_indexed_by(m.b.v2, m.space))
+
+        self.assertFalse(is_implicitly_indexed_by(m.v1, m.time))
+        self.assertFalse(is_implicitly_indexed_by(m.v2, m.set))
+        self.assertTrue(is_implicitly_indexed_by(m.b1[m.time[1]].v2, m.time))
+
+        self.assertTrue(is_implicitly_indexed_by(m.b2[m.time[1], 
+            m.space[1]].b.v1, m.time))
+        self.assertEqual(is_implicitly_indexed_by(m.b2[m.time[1], 
+            m.space[1]].b.v2, m.time),
+            is_explicitly_indexed_by(m.b2[m.time[1], 
+                m.space[1]].b.v2, m.time))
+        self.assertFalse(is_implicitly_indexed_by(m.b2[m.time[1], 
+            m.space[1]].b.v1, m.set))
+
+        self.assertFalse(is_implicitly_indexed_by(m.b2[m.time[1],
+            m.space[1]].b.v1, m.space, stop_at=m.b2[m.time[1], m.space[1]]))
+
+
+    # Test get_index_set_except and _complete_index
+    def test_get_index_set_except(self):
+        '''
+        Tests:
+          For components indexed by 0, 1, 2, 3, 4 sets:
+            get_index_set_except one, then two (if any) of those sets
+            check two items that should be in set_except
+            insert item(s) back into these sets via index_getter
+        '''
+        m = ConcreteModel()
+        m.time = ContinuousSet(bounds=(0, 10))
+        m.space = ContinuousSet(bounds=(0, 10))
+        m.set1 = Set(initialize=['a', 'b', 'c'])
+        m.set2 = Set(initialize=['d', 'e', 'f'])
+        m.v = Var()
+        m.v1 = Var(m.time)
+        m.v2 = Var(m.time, m.space)
+        m.v3 = Var(m.time, m.space, m.set1)
+        m.v4 = Var(m.time, m.space, m.set1, m.set2)
+
+        disc = TransformationFactory('dae.collocation')
+        disc.apply_to(m, wrt=m.time, nfe=5, ncp=2, scheme='LAGRANGE-RADAU')
+        disc.apply_to(m, wrt=m.space, nfe=5, ncp=2, scheme='LAGRANGE-RADAU')
+
+        # Want this to give a TypeError
+        # info = get_index_set_except(m.v, m.time)
+
+        # Indexed by one set
+        info = get_index_set_except(m.v1, m.time)
+        set_except = info['set_except']
+        index_getter = info['index_getter']
+        self.assertTrue(set_except == [None])
+        self.assertEqual(index_getter([5,6,7], 3j), 3j)
+
+        # Indexed by two sets
+        info = get_index_set_except(m.v2, m.time)
+        set_except = info['set_except']
+        index_getter = info['index_getter']
+        self.assertTrue(m.space[1] in set_except
+                        and m.space.last() in set_except)
+        self.assertEqual(index_getter(2, 4), (4, 2), index_getter((2,), 4))
```

Comment:
What are you testing here?  it looks like you are providing `index_getter((2,),4)` the failure message?  If that is your intent, please at lease form a string message that would help a developer to understand what went wrong.

Reply:
OK - your comment helps.  However, that should be two separate `assertEqual` statements (one for `(2,4)` and one for `((2,),4)`).  Right now, you are providing `index_getter((2,), 4)` as the `message` argument to `assertEqual`.